stpsf/roman.py

```diff
@@ -779,6 +858,63 @@ def aperturename(self, value):
+        for fl, info in self._filters.items():
+            filter_path = Path(info.filename)
+
+            # find 'filters', the earliest common parent directory of default
+            # stpsf filter file structure (stpsf-data/INSTRUMENT/*filters*)
+            # and WFI's filter file structure (stpsf-data/WFI/*filters*/DETECTOR).
+            if not found_parent:
+                for parent in filter_path.parents:
+                    if parent.name == 'filters':
+                        found_parent = True
+                        break
+                    else:
+                        continue
+            # done this way because SpaceTelescopeInstrument.__init__()
+            # (a WFI parent class) sets filter location to the default WFI path
+            # with SpaceTelescopeInstrument._get_filters() upon class creation,
+            # so the filter files' direct parent directory the first time a
+            # detector is assigned in WFI.__init__() (the "filters" directory)
+            # is not at the same level as on later changes to the detector attribute
+            # (the corresponding "WFINN" child directory of "WFI/filters").
+
+            # save filter info but change throughput file based on new detector
+            info_edit = {key : (val if key != 'filename'
+                                else str(parent / self._detector / filter_path.name))
+                         for key, val in info._asdict().items()}
+            filters_info[fl] = stpsf_core.Filter(**info_edit)
+
+        self._filters = filters_info
+
+    ################################
+    # FILTER THROUGHPUT NOTE:
+    # could also supersede SpaceTelescopeInstrument._get_filters() with a
+    # _get_filters() in RomanInstrument or WFI that adds a "detector" to
+    # the filter_filename path, but we don't yet have WFI._detector_list
+    # [set in WFI._load_detector_aberrations() later in WFI.__init__()]
+    # or WFI._detector [set through WFI.detector later in WFI.__init__()].
+
+    # {A} Can't set WFI._detector_list earlier in WFI.__init__() because
+    # generating the path needed as an argument requires knowing the
+    # filters that the new _get_filters() method is trying to generate
+    # in the first place.
+
+    # {B} Can't set WFI.detector earlier in
+    # WFI.__init__() because SpaceTelescopeInstrument.__init__() will
+    # set self._detector back to None. Even a modification to
+    # SpaceTelescopeInstrument.__init__() like
+    # `if not hasattr(self, '_detector'): self._detector = None`
+    # (which I'm wary to do due to unintended effects on the JWST classes)
+    # because WFI.detector's setter triggers WFI._update_pupil(), which
+    # again requires knowledge of the filters. COULD do this if we set
+    # WFI._detector (note underscore) instead, but do we want to skip the
+    # setter in that scenario?
+    ################################
+
     @RomanInstrument.filter.setter
     def filter(self, value):
```

Comment:
The string for F146 is now shows as W146 in filter_list. The filename for the throughput file in the new STPSF data also shows W146. This should be changed back to F146.

Reply:
The filter is relabeled as W146 across the spreadsheet, pupil images, etc. delivered from GSFC. I hadn't seen W146 before, but it seemed like a lot of occurrences to be a typo. Do you know where it might come from?

Reply:
I think this is a relic of earlier notation for the different filters... All of our current documentation, including Goddard's, refers to this filter as F146.  APT and the ETC also use F146.

Reply:
The file `WIM_W146_zernikes_cycle10.csv` will also need to be renamed.

Reply:
For completeness, this also involves renaming the filter's entry in `filters.tsv`. I have reverted to the original filter name in all of these places and copied the updated reference data to the same location as earlier.